python/mp/rosary.py

```diff
@@ -57,9 +87,29 @@ def __init__(self, ip="127.0.0.1", port=5005, dispatcher=None, name="rosary"):
         self.dispatcher = dispatcher
         # Available knobs to turn
         self.knobs = {}
+        self.updater_list = []
 
+        self.osc_client = udp_client.UDPClient(self.osc_ip, self.osc_port)
+
+        # create the three classes of LED "beads"
         for i in range(self.BEAD_COUNT):
             self.beads.append(Bead(i))
+        self.updater_list.append(Updater(name='rosary',
+                                         bead_list=self.beads,
+                                         osc_client=self.osc_client))
+
+        for i in range(self.BASE_COUNT):
+            self.bases.append(Bead(i))
+        self.updater_list.append(Updater(name='base',
+                                         bead_list=self.bases,
+                                         osc_client=self.osc_client))
+
+        for i in range(self.CROSS_LED_COUNT):
+            self.cross.append(Bead(i))
+        self.updater_list.append(Updater(name='cross',
+                                         bead_list=self.cross,
+                                         osc_client=self.osc_client))
+
```

Comment:
No, Bead should not take a name. self.bases[0], self.beads[0] and self.cross[0] do not refer to the same bead. There are 3 python lists of beads to help us keep them straight, but the effects never see these arrays. The effects get sets of beads, and we can pull those sets from any of the three sets of beads.

Right now if you mix beads from multiple lists in the same set we have a problem b/c they won't sort correctly. I'm "sorting" that out still. Hah. In reality I don't think we will want to combine, say, the cross and the rosary in the same effect so I'm not too concerned.